### PR TITLE
Bug fix for thermostat application for wifi 

### PR DESCRIPTION
**Pull request by kiran0284**
Added :
    -   added fix for OccupiedCoolingSetpoint and MaxHeatSetpointLimit write command in thermostat cluster

Issue :
    -   while trying set OccupiedCoolingSetpoint and MaxHeatSetpointLimit got CONSTRAINT_ERROR instead of SUCCESS

Tested  :
    -   manually tested on MG24 + 917 Exp

### CODE REVIEW

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 1]
```diff
@@ -349,11 +349,11 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
         requested = static_cast<int16_t>(chip::Encoding::LittleEndian::Get16(value));
         if (!HeatSupported)
             return imcode::UnsupportedAttribute;
-        if (requested < AbsMinHeatSetpointLimit || requested < MinHeatSetpointLimit || requested > AbsMaxHeatSetpointLimit)
+        if (requested < AbsMinHeatSetpointLimit || requested < MinHeatSetpointLimit || requested < AbsMaxHeatSetpointLimit)
```
Comment (mkardous-silabs) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:352]: This change doesn't seem correct. Shouldn't we return `InvalidValue` if the requested value is superior to the `AbsMaxHeatSetpointLimit` ?

**Comment by mkardous-silabs** [comment]
@kiran0284 Could you change the name of the PR by removing the `[Silabs] [EFR32]` part? The changes you are proposing of not platform specific. The name is misleading in this particular case.

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 2]
```diff
@@ -297,7 +297,7 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
             return imcode::InvalidValue;
         if (AutoSupported)
         {
-            if (requested < OccupiedHeatingSetpoint + DeadBandTemp)
+            if (requested > OccupiedHeatingSetpoint + DeadBandTemp)
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:300]: This change is wrong, as far as I can tell.  What the spec says is:

> if, and only if, the AUTO feature is supported, a deadband must be maintained between Heating and Cooling setpoints and limits:
> ...
> OccupiedHeatingSetpoint ≤ (OccupiedCoolingSetpoint - MinSetpointDeadBand)

So if that condition is violated, i.e. "OccupiedHeatingSetpoint + MinSetpointDeadBand > OccupiedCoolingSetpoint", then that's not a valid setting.  And that's what this code is checking for.

Please clearly explain why you are introducing these changes that violate the spec (and incidentally fail the conformance tests like Test_TC_TSTAT_2_2).

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Comment by andy31415** [comment]
1 year old PR is in draft, has changes requested and merge conflicts. Closing as stale.

**Pull request closed without merging**